### PR TITLE
[matrix] One-Page-Per-View Emulator

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

> :warning: **This is extremely W.I.P. and needs to be optimised. Code reviews welcomed.**

Implements a discord.js-style documentation where each class is on a separate *page*.

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/_static/pages.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+function collapseClasses() {
+  let sections = document.querySelectorAll(
+    "div.section:not(#api-reference)"
+  );
+  for (let sect of sections) {
+    let classes = Array.from(sect.children).filter((el) =>
+      el.classList.contains("section")
+    );
+
+    classes.forEach((v, i, a) => {
```
Comment (slice) [inline docs/_static/pages.js:110]: You can drop `a` here.

**Comment by slice** [comment]
I didn't review some instances of these:

- `Array.from(Node).filter` can usually be reduced into `Node.querySelector`.
- `Array.from(Node).forEach` can be reduced into `for (const Subnode of Node)`

File: docs/_static/custom.js [hunk 2]
```diff
@@ -39,20 +41,36 @@ function getRootAttributeToggle(attributeName, valueName) {
...
+    }
+  });
+}
+
 const settings = {
-  useSansFont: getRootAttributeToggle('font', 'sans'),
-  useDarkTheme: getRootAttributeToggle('theme', 'dark')
+  useSansFont: [getRootAttributeToggle('font', 'sans'), true],
+  useDarkTheme: [getRootAttributeToggle('theme', 'dark'), true],
+  useClassicAPIRef: [toggleClassicAPIRef, false]
```
Comment (Rapptz) [inline docs/_static/custom.js:70]: Definitely not a fan of this. It seems weird. I don't like random booleans.

File: docs/_static/custom.js [hunk 3]
```diff
@@ -39,20 +41,36 @@ function getRootAttributeToggle(attributeName, valueName) {
...
+  if (apiReference) {
+    var event = new CustomEvent('APIRefStyle', { detail: { classic: add } });
+    window.dispatchEvent(event);
+    enableScroll = add;
+  } else {
+    enableScroll = true;
+  }
+}
+
+function loadSettings(settingsDict, pre=true) {
```
Comment (Rapptz) [inline docs/_static/custom.js:54]: Why is thsi `pre` argument even here?

File: docs/_static/custom.js [hunk 4]
```diff
@@ -39,20 +41,36 @@ function getRootAttributeToggle(attributeName, valueName) {
   return toggleRootAttribute;
 }
 
+function toggleClassicAPIRef(add) {
+  if (apiReference) {
+    var event = new CustomEvent('APIRefStyle', { detail: { classic: add } });
```
Comment (Rapptz) [inline docs/_static/custom.js:46]:
```suggestion
    let event = new CustomEvent('APIRefStyle', { detail: { classic: add } });
```

What is this custom event for anyway?

File: docs/conf.py [hunk 5]
```diff
@@ -237,6 +237,7 @@
 html_js_files = [
   'custom.js',
   'copy.js',
+  'pages.js'
```
Comment (Rapptz) [inline docs/conf.py:240]:
```suggestion
  'pages.js',
```

File: docs/_static/pages.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+      expandAll();
+    }
+    obj = resolveClassicHref(window.location.href) || apiReference;
+  } else {
+    collapseSections();
+    collapseClasses();
+  
+    obj = resolveHref(window.location.href);
+  }
+  apiReference.style.display = "block";
```
Comment (Rapptz) [inline docs/_static/pages.js:16]: Cross-script variables are a very very bad practice.

**Pull request closed without merging**